spicy/lib/spicy_rt.hlt

```diff
@@ -33,11 +33,11 @@ public type HiltiResumable = __library_type("hilti::rt::Resumable");
 public type Filters = __library_type("spicy::rt::filter::detail::Filters");
 public type Forward = __library_type("spicy::rt::filter::detail::Forward");
 
-declare public strong_ref<stream> filter_init(inout any unit, inout value_ref<stream> data, view<stream> cur) &cxxname="spicy::rt::filter::init" &have_prototype;
-declare public void filter_connect(inout any unit, strong_ref<*> filter) &cxxname="spicy::rt::filter::connect" &have_prototype;
-declare public void filter_disconnect(inout any unit) &cxxname="spicy::rt::filter::disconnect" &have_prototype;
-declare public void filter_forward(inout any filter, bytes b) &cxxname="spicy::rt::filter::forward" &have_prototype;
-declare public void filter_forward_eod(inout any filter) &cxxname="spicy::rt::filter::forward_eod" &have_prototype;
+declare public strong_ref<stream> filter_init(inout any unit &requires_type="supports_filters", inout value_ref<stream> data, view<stream> cur) &cxxname="spicy::rt::filter::init" &have_prototype;
+declare public void filter_connect(inout any unit &requires_type="supports_filters", strong_ref<*> filter &requires_type="is_filter") &cxxname="spicy::rt::filter::connect" &have_prototype;
+declare public void filter_disconnect(inout any unit &requires_type="supports_filters") &cxxname="spicy::rt::filter::disconnect" &have_prototype;
+declare public void filter_forward(inout any filter &requires_type="is_filter", bytes b) &cxxname="spicy::rt::filter::forward" &have_prototype;
+declare public void filter_forward_eod(inout any filter &requires_type="is_filter") &cxxname="spicy::rt::filter::forward_eod" &have_prototype;
```

Comment:
We should annotate additional function declarations so we can remove the workaround we added in 6160ef603bfa285dccc4ca52bad2d9e422cd1078 which leads us to always emit parser functions for filters, even if unused. Above functions are not sufficient for that.

Reply:
I'll move filter-related work to a follow-up PR.